spec/realtime/failure.test.js

```diff
@@ -25,13 +25,15 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
 	 * Connect with invalid credentials on various transports; connection state should be 'failed'
 	 */
 	exports.invalid_cred_failure = function(test) {
-		test.expect(availableTransports.length + 1);
+		test.expect((availableTransports.length + 1)*3);
 		try {
 			var failure_test = function(transports) {
 				return function(cb) {
 					var realtime = helper.AblyRealtime({key: "this.is:wrong", transports: transports});
-					realtime.connection.on('failed', function() {
+					realtime.connection.on('failed', function(connectionStateChange) {
 						test.ok(true, 'connection state for ' + transports + ' was failed, as expected');
+						test.notEqual(realtime.connection.errorReason, null, 'error reason was not set on connection');
+						test.notEqual(connectionStateChange.reason, null, 'error reason was not set on connectionStateChange');
```

Comment:
@adamgiacomelli looks fine as far as it goes, but would be even better to
- test that the `reason` is actually an ErrorInfo object as it should be and shows the expected code for that kind of failure, eg `test.equal(reason.code, 40400)`, rather than just not being null, and
- test that the `connection.errorReason` and the `connectionStateChange.reason` are `deepEqual` to each other, rather than just testing that they both exist